Civi/Core/SettingsMetadata.php

```diff
@@ -145,20 +151,41 @@ protected static function loadSettingsMetadata($metaDataFolder) {
    *   Metadata to filter.
    */
   protected static function _filterSettingsSpecification($filters, &$settingSpec) {
-    if (empty($filters)) {
-      return;
-    }
-    elseif (array_keys($filters) == ['name']) {
-      $settingSpec = [$filters['name'] => \CRM_Utils_Array::value($filters['name'], $settingSpec, '')];
-      return;
+    if (!empty($filters['name'])) {
+      $settingSpec = array_intersect_key($settingSpec, array_flip((array) $filters['name']));
+      // FIXME: This is a workaround for settingsBag::setDb() called by unit tests with settings names that don't exist
+      $settingSpec += array_fill_keys((array) $filters['name'], []);
+      unset($filters['name']);
     }
-    else {
+    if (!empty($filters)) {
       foreach ($settingSpec as $field => $fieldValues) {
         if (array_intersect_assoc($fieldValues, $filters) != $filters) {
           unset($settingSpec[$field]);
         }
       }
-      return;
+    }
+  }
+
+  /**
+   * Retrieve options from settings metadata
+   *
+   * @param array $settingSpec
+   */
+  protected static function _loadOptions(&$settingSpec) {
```

Comment:
so on a style thing - my understanding is we've been moving away from the leading underscore on protected functions (and in fact it's inconsistent within the class)

Reply:
Agreed. I've updated the style thing.